pkg/operator/controller/ingress/status.go

```diff
@@ -659,12 +680,20 @@ func computeIngressProgressingCondition(conditions []operatorv1.OperatorConditio
 				case configv1.AWSPlatformType, configv1.IBMCloudPlatformType:
 					message = fmt.Sprintf("%[1]s  To effectuate this change, you must delete the service: `oc -n %[2]s delete svc/%[3]s`; the service load-balancer will then be deprovisioned and a new one created.  This will most likely cause the new load-balancer to have a different host name and IP address from the old one's.  Alternatively, you can revert the change to the IngressController: `oc -n openshift-ingress-operator patch ingresscontrollers/%[4]s --type=merge --patch='{\"spec\":{\"endpointPublishingStrategy\":{\"loadBalancer\":{\"scope\":\"%[5]s\"}}}}'", message, service.Namespace, service.Name, ic.Name, haveScope)
 				}
-				condition.Reason = "ScopeChanged"
-				condition.Message = message
-				condition.Status = operatorv1.ConditionTrue
+				progressingReasons = append(progressingReasons, "ScopeChanged")
+				messages = append(messages, message)
 			}
 		}
 	}
+	if len(progressingReasons) != 0 {
+		// Let ConditionUnknown status take precedence over ConditionTrue if it was set by logic above.
+		if condition.Status != operatorv1.ConditionUnknown {
+			condition.Status = operatorv1.ConditionTrue
+		}
+		condition.Reason = strings.Join(progressingReasons, "")
```

Comment:
Like I said, I copied from [here](https://github.com/openshift/cluster-dns-operator/blob/d50df32df68f53c1d47db8f5e51a8b27c402f278/pkg/operator/controller/dns_status.go#L221.). Should I join with "," since that might look cleaner or keep this consistent with DNS Operator like I have it now?

Reply:
`Reason` shouldn't have punctuation or spaces, so joining using `""` is appropriate.

Reply:
You could join on `"And"` so you'd get "ReconcilingAndIncompleteStatus" instead of "ReconcilingIncompleteStatus".  Both "Reconciling" and "IncompleteStatus" seem too vague though; they could be changed to "DeploymentProgressing" and "LoadBalancerPending" or similar, respectively.